### PR TITLE
Consume upjet with custom metrics

### PR DESCRIPTION
**Pull request by turkenf**
### Description of your changes

This PR bumps the upjet dependency to make the new custom Prometheus metrics available in the provider's runtime. Please see https://github.com/upbound/upjet/pull/170 for the available metrics and examples.

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
Please see https://github.com/upbound/upjet/pull/170.

### CODE REVIEW

File: cmd/provider/main.go [hunk 1]
```diff
@@ -44,14 +44,16 @@ import (
...
+		pollInterval       = app.Flag("poll", "Poll interval controls how often an individual resource should be checked for drift.").Default("10m").Duration()
+		leaderElection     = app.Flag("leader-election", "Use leader election for the controller manager.").Short('l').Default("false").OverrideDefaultFromEnvar("LEADER_ELECTION").Bool()
+		terraformVersion   = app.Flag("terraform-version", "Terraform version.").Required().Envar("TERRAFORM_VERSION").String()
+		providerSource     = app.Flag("terraform-provider-source", "Terraform provider source.").Required().Envar("TERRAFORM_PROVIDER_SOURCE").String()
+		providerVersion    = app.Flag("terraform-provider-version", "Terraform provider version.").Required().Envar("TERRAFORM_PROVIDER_VERSION").String()
+		maxReconcileRate   = app.Flag("max-reconcile-rate", "The global maximum rate per second at which resources may checked for drift from the desired state.").Default("10").Int()
+		nativeProviderPath = app.Flag("terraform-native-provider-path", "Terraform native provider path for shared execution.").Default("").Envar("TERRAFORM_NATIVE_PROVIDER_PATH").String()
```
Comment (ulucinar) [inline cmd/provider/main.go:56]: We do not aim to enable the shared gRPC runtime with this PR. So, let's not introduce this command-line flag with this PR.

File: cmd/provider/main.go [hunk 2]
```diff
@@ -83,6 +85,10 @@ func main() {
 	})
 	kingpin.FatalIfError(err, "Cannot create controller manager")
 	kingpin.FatalIfError(apis.AddToScheme(mgr.GetScheme()), "Cannot add Azuread APIs to scheme")
+	var runner terraform.ProviderRunner = terraform.NewNoOpProviderRunner()
+	if len(*nativeProviderPath) != 0 {
+		runner = terraform.NewSharedProvider(log, *nativeProviderPath, "registry.terraform.io/"+*providerSource)
+	}
```
Comment (ulucinar) [inline cmd/provider/main.go:91]: Let's exclude the shared gRPC server related setup here.

File: cmd/provider/main.go [hunk 3]
```diff
@@ -93,7 +99,7 @@ func main() {
 		Provider: config.GetProvider(),
 		// use the following WorkspaceStoreOption to enable the shared gRPC mode
 		// terraform.WithProviderRunner(terraform.NewSharedProvider(log, os.Getenv("TERRAFORM_NATIVE_PROVIDER_PATH"), terraform.WithNativeProviderArgs("-debuggable")))
-		WorkspaceStore: terraform.NewWorkspaceStore(log),
+		WorkspaceStore: terraform.NewWorkspaceStore(log, terraform.WithProviderRunner(runner), terraform.WithProcessReportInterval(*pollInterval)),
```
Comment (ulucinar) [inline cmd/provider/main.go:102]:
```suggestion
		WorkspaceStore: terraform.NewWorkspaceStore(log, terraform.WithProcessReportInterval(*pollInterval)),
```

File: cmd/provider/main.go [hunk 4]
```diff
@@ -47,6 +47,7 @@ func main() {
 		app              = kingpin.New(filepath.Base(os.Args[0]), "Terraform based Crossplane provider for Azuread").DefaultEnvars()
 		debug            = app.Flag("debug", "Run with debug logging.").Short('d').Bool()
 		syncPeriod       = app.Flag("sync", "Controller manager sync period such as 300ms, 1.5h, or 2h45m").Short('s').Default("1h").Duration()
+		pollInterval     = app.Flag("poll", "Poll interval controls how often an individual resource should be checked for drift.").Default("10m").Duration()
```
Comment (ulucinar) [inline cmd/provider/main.go:50]: Let's also add the [`--max-reconcile-rate` option](https://github.com/upbound/provider-aws/blob/025f49bfecf5c4a1bcd5ac2cb9cb6de007be70ea/cmd/provider/main.go#L43) here, and pass them to the crossplane-runtime controller options.

**Comment by turkenf** [comment]
/test-examples="examples/applications/application.yaml"

**Pull request merged**